### PR TITLE
Module5

### PR DESCRIPTION
**Pull request by SerhiiSmilianets**
2024-03-08_11h04_52
2024-03-08_11h05_06

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,19 +1,9 @@
...
-import Counter from './components/Counter';
-import SearchBar from './components/SearchBar';
-import GenreList from './components/GenreList';
-import MoviesList from './components/MoviesList';
-import DialogOpenButton from './components/DialogOpenButton';
+import MovieListPage from './pages/MovieListPage'
 
 function App() {
   return (
     <>
```
Comment (MykhailoSheliahov) [inline src/App.js]: ✂

File: src/components/Dialog.tsx [hunk 2]
```diff
@@ -1,11 +1,6 @@
...
-import '../components-styles/Dialog.scss';
-
-interface DialogProps {
-  title: string;
-  handleClose: () => void;
-  children: ReactNode;
-}
+import React, { MouseEvent } from 'react';
+import '../styles/Dialog.scss';
+import {DialogProps} from '../types'
```
Comment (MykhailoSheliahov) [inline src/components/Dialog.tsx]: Props for component should be defined inside component

File: src/components/MovieForm.tsx [hunk 3]
```diff
@@ -1,22 +1,7 @@
...
-  runtime: number;
-  overview: string;
-}
-
-interface MovieFormProps {
-  movieData?: MovieData;
-  onSubmit: (event: FormEvent<HTMLFormElement>) => void;
-  isDeleteForm: boolean;
-}
+import {MovieFormProps} from '../types'
```
Comment (MykhailoSheliahov) [inline src/components/MovieForm.tsx]: MovieFormProps type  should be defined inside component, but type for  movieData should be imported from types file

File: src/pages/MovieListPage.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect } from "react";
+
+//components import
```
Comment (MykhailoSheliahov) [inline src/pages/MovieListPage.tsx:3]: ✂

File: src/pages/MovieListPage.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+import SearchBar from '../components/SearchBar';
+import GenreList from '../components/GenreList';
+import SortControl from "../components/SortControl";
+import MovieTile from "../components/MovieTile";
+import MovieDetails from "../components/MovieDetails";
+import DialogOpenButton from '../components/DialogOpenButton';
+
+//styles import
```
Comment (MykhailoSheliahov) [inline src/pages/MovieListPage.tsx:11]: ✂

File: src/pages/MovieListPage.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+import '../styles/MovieListPage.scss';
+//helpers import
```
Comment (MykhailoSheliahov) [inline src/pages/MovieListPage.tsx:13]: ✂

File: src/pages/MovieListPage.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+import {GENRE_LIST} from '../constants'
+//types import
```
Comment (MykhailoSheliahov) [inline src/pages/MovieListPage.tsx:15]: ✂

File: src/components/GenreList.jsx [hunk 8]
```diff
@@ -10,18 +9,12 @@ const GenreBtn = ({genre, isSelected, onGenreSelection}) => {
...
 }
 
-const GenreList = ({ genresList }) => {
-    const [selectedGenre, setSelectedGenre] = useState('');
-
-    const onSelect = (genre) => {
-        setSelectedGenre(genre)
-    }
-
+const GenreList = ({ genresList, selectedGenre, onSelectGenre }) => {
```
Comment (MykhailoSheliahov) [inline src/components/GenreList.jsx]: Check also in other places
```suggestion
const GenreList = ({ genresList, selectedGenre, handleSelectGenre }) => {
```

File: src/components/MovieDetails.jsx [hunk 9]
```diff
@@ -1,9 +1,9 @@
 import PropTypes from 'prop-types';
 import { getRuntimeFormatted, getReleaseYear } from '../utils/movieUtils';
 
-import '../components-styles/MovieDetails.scss';
+import '../styles/MovieDetails.scss';
 
-const MovieDetails = ({poster_path, title, vote_average, genres, release_date, runtime, overview}) => {
+const MovieDetails = ({poster_path, title, vote_average, genres, release_date, runtime, overview, resetHeader}) => {
```
Comment (MykhailoSheliahov) [inline src/components/MovieDetails.jsx:6]:
```suggestion
const MovieDetails = ({posterPath, title, voteAverage, genres, releaseDate, runtime, overview, resetHeader}) => {
```

File: src/components/MovieForm.tsx [hunk 10]
```diff
@@ -1,22 +1,7 @@
...
 
 const MovieForm: React.FC<MovieFormProps> = ({movieData, onSubmit, isDeleteForm}) => {
```
Comment (MykhailoSheliahov) [inline src/components/MovieForm.tsx]:
```suggestion
const MovieForm: React.FC<MovieFormProps> = ({movieData, handleSubmit, isDeleteForm}) => {
```

File: src/components/MovieTile.jsx [hunk 11]
```diff
@@ -1,24 +1,67 @@
...
+            if (menuRef.current && !menuRef.current.contains(event.target)) {
+                setMenuTileContextOpen(false)
+            }
         }
+
+        document.addEventListener("mousedown", handleClickOutside);
+        return () => {
+
+            document.removeEventListener("mousedown", handleClickOutside);
+        };
```
Comment (MykhailoSheliahov) [inline src/components/MovieTile.jsx:19]:
```suggestion
        return () => document.removeEventListener("mousedown", handleClickOutside)
```

File: src/components/MovieTile.jsx [hunk 12]
```diff
@@ -1,24 +1,67 @@
...
+    }, [menuRef, setMenuTileContextOpen]);
+}
+
+const MovieTile = ({movieData, onClick}) => {
+    const [isMenuTileContextOpen, setMenuTileContextOpen] = useState(false);// need to hide/show menu
```
Comment (MykhailoSheliahov) [inline src/components/MovieTile.jsx]:
```suggestion
    const [isMenuOpen, setIsMenuOpen] = useState(false);// need to hide/show menu
```

File: src/components/MovieTile.jsx [hunk 13]
```diff
@@ -1,24 +1,67 @@
...
+    const [isMenuTileContextRendered, setMenuTileContextRendered] = useState(false)// need for menu render
```
Comment (MykhailoSheliahov) [inline src/components/MovieTile.jsx:25]: Can we remove it and use setIsMenuOpen?

File: src/components/MovieTile.jsx [hunk 14]
```diff
@@ -1,24 +1,67 @@
...
 import { getReleaseYear } from '../utils/movieUtils';
-import '../components-styles/MovieTile.scss';
+import '../styles/MovieTile.scss';
+import DialogOpenButton from '../components/DialogOpenButton';
 
-const MovieTile = ({poster_path, title, genres, release_date, onClick}) => {
-    const handleClickEvent = (e) => {
-        if (typeof onClick === "function") {
-            onClick(e.currentTarget.value)
+const useOutsideClick = (menuRef, setMenuTileContextOpen) => {
```
Comment (MykhailoSheliahov) [inline src/components/MovieTile.jsx:7]: Lets create this  setMenuTileContextOpen state  value inside hook and return it value as boolean

File: src/components/SortControl.jsx [hunk 15]
```diff
@@ -1,16 +1,11 @@
...
 import PropTypes from 'prop-types';
 
-const SortControl = ({selectedSorting, onChange}) => {
-    const handleOnChange = (e) => {
-        if (typeof onChange === "function") {
-            onChange(e.currentTarget.value)
-        }
-    }
-
+const SortControl = ({selectedSorting, onSortChange}) => {
```
Comment (MykhailoSheliahov) [inline src/components/SortControl.jsx]:
```suggestion
const SortControl = ({selectedSorting, handleSortChange}) => {
```

File: src/components/SortControl.jsx [hunk 16]
```diff
@@ -1,16 +1,11 @@
...
     return (
         <div className="sorting-container">
             <p>SORT BY</p>
-            <select value={selectedSorting} onChange={handleOnChange}>
+            <select value={selectedSorting} onChange={(e) => onSortChange(e.currentTarget.value)}>
```
Comment (MykhailoSheliahov) [inline src/components/SortControl.jsx]:
```suggestion
            <select value={selectedSorting} onChange={(e) => handleSortChange(e.currentTarget.value)}>
```

File: src/pages/MovieListPage.tsx [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+    useEffect(() => {
+        const getMovies = async () => {
+            const response = await fetch('http://localhost:4000/movies');
+            const resData = await response.json();
+            setMoviesList(resData.data);
+        }
+        getMovies()
+    }, [])
+
+    const onSearch = (value: string) => {
```
Comment (MykhailoSheliahov) [inline src/pages/MovieListPage.tsx]:
```suggestion
    const handleSearch = (value: string) => {
```

File: src/pages/MovieListPage.tsx [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+                    ?
+                    <MovieDetails 
+                        poster_path={selectedMovieData.poster_path} 
+                        title={selectedMovieData.title} 
+                        vote_average={selectedMovieData.vote_average} 
+                        genres={selectedMovieData.genres} 
+                        release_date={selectedMovieData.release_date} 
+                        runtime={selectedMovieData.runtime} 
+                        overview={selectedMovieData.overview}
+                        resetHeader={setSelectedMovieData}
```
Comment (MykhailoSheliahov) [inline src/pages/MovieListPage.tsx:51]:
```suggestion
                        posterPath={selectedMovieData.poster_path} 
                        title={selectedMovieData.title} 
                        voteAverage={selectedMovieData.vote_average} 
                        genres={selectedMovieData.genres} 
                        releaseDate={selectedMovieData.release_date} 
                        runtime={selectedMovieData.runtime} 
                        overview={selectedMovieData.overview}
                        resetHeader={setSelectedMovieData}
```

**Review by MykhailoSheliahov — CHANGES_REQUESTED** [review]

File: src/pages/MovieListPage.tsx (hunk 4 above)
Comment (SerhiiSmilianets) [reply src/pages/MovieListPage.tsx:3]: comment to mark imports, why to remove?

File: src/components/MovieTile.jsx (hunk 13 above)
Comment (SerhiiSmilianets) [reply src/components/MovieTile.jsx:25]: @MykhailoSheliahov no, there is an issue with the context menu. This context menu contains buttons that handle portals. We can have only one to control hidden class.

File: src/components/MovieTile.jsx (hunk 14 above)
Comment (SerhiiSmilianets) [reply src/components/MovieTile.jsx:7]: @MykhailoSheliahov there is also an event on button "close" that changes the state

**Review by MykhailoSheliahov — APPROVED** [review]

**Pull request merged**